shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -1000,15 +1020,11 @@ trait StatementImpl extends Located { self: Statement =>
     case AdtMatchWith(cond, _) => cond :: Nil // FIXME discards branches...
   }
   
-  
-  override def toString: Str = this match {
-    case LetS(isRec, name, rhs) => s"let${if (isRec) " rec" else ""} $name = $rhs"
-    case DatatypeDefn(head, body) => s"data type $head of $body"
-    case DataDefn(head) => s"data $head"
-    case Constructor(params, bod) => s"constructor(${params.showElems}) $bod"
-    case _: Term => super.toString
-    case d: Decl => d.showDbg
-    case d: NuDecl => d.showDbg
+  def showDbg: Str = this match {
+    case LetS(isRec, name, rhs) => s"let${if (isRec) " rec" else ""} ${name.showDbg} = ${rhs.showDbg}"
+    case DatatypeDefn(head, body) => s"data type ${head.showDbg} of ${body.showDbg}"
+    case DataDefn(head) => s"data ${head.showDbg}"
+    case _: Constructor | _: Term | _: Decl | _: NuDecl => ??? // Implemented in subclasses
   }
 }
```

Comment:
`LetS`, `DatatypeDefn`, and `Datadefn` doesn't have implementation traits. So this `showDbg` at `StatementImpl` must have a definition. I left an `???` because `Constructor`, `Term`, `Decl`, and `NuDecl` has their implementation traits. Is this okay?

Reply:
Of course not. This code is dangerous and not refactoring-safe. Just move the implementations here.

Reply:
In general you should never make the logic of a piece of code depend on some implementation detail in another piece of code located somewhere else in the code base. It's still needed sometimes (not here), and then it needs extensive documentation.

Reply:
Thanks! I will fix this immediately.